[BUG]: Can't compile if I add #include "LcdMenu.h" to any other file than main.cpp

### What happened?

in my platformio project, if I add #include <LcdMenu.h>to another .h file, I get a conflict error when compiling.

It appears to only happen if the .h is included in a file that contains a constructor with at least 1 argument. Very strange.

Error:

```
c:/users/daniel/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: .pio\build\main\lib4fd\libmenu.a(MenuClass.cpp.o): in function `substring(char*, unsigned char, unsigned char, char*)':
D:\9mediatech\acrobotics\code\jona\Acrobot/.pio/libdeps/main/LcdMenu/src/utils.h:4: multiple definition of `substring(char*, unsigned char, unsigned char, char*)'; .pio\build\main\src\main.cpp.o:D:\9mediatech\acrobotics\code\jona\Acrobot/.pio/libdeps/main/LcdMenu/src/utils.h:4: first defined here
c:/users/daniel/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: .pio\build\main\lib4fd\libmenu.a(MenuClass.cpp.o): in function `concat(char const*, char, char const*, char*)':
D:\9mediatech\acrobotics\code\jona\Acrobot/.pio/libdeps/main/LcdMenu/src/utils.h:9: multiple definition of `concat(char const*, char, char const*, char*)'; .pio\build\main\src\main.cpp.o:D:\9mediatech\acrobotics\code\jona\Acrobot/.pio/libdeps/main/LcdMenu/src/utils.h:9: first defined here
c:/users/daniel/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: .pio\build\main\lib4fd\libmenu.a(MenuClass.cpp.o): in function `concat(char const*, char, char*)':
D:\9mediatech\acrobotics\code\jona\Acrobot/.pio/libdeps/main/LcdMenu/src/utils.h:19: multiple definition of `concat(char const*, char, char*)'; .pio\build\main\src\main.cpp.o:D:\9mediatech\acrobotics\code\jona\Acrobot/.pio/libdeps/main/LcdMenu/src/utils.h:19: first defined here
c:/users/daniel/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: .pio\build\main\lib4fd\libmenu.a(MenuClass.cpp.o): in function `concat(char const*, char const*, char*)':
D:\9mediatech\acrobotics\code\jona\Acrobot/.pio/libdeps/main/LcdMenu/src/utils.h:26: multiple definition of `concat(char const*, char const*, char*)'; .pio\build\main\src\main.cpp.o:D:\9mediatech\acrobotics\code\jona\Acrobot/.pio/libdeps/main/LcdMenu/src/utils.h:26: first defined here
c:/users/daniel/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: .pio\build\main\lib4fd\libmenu.a(MenuClass.cpp.o): in function `remove(char*, unsigned char, unsigned char)':
D:\9mediatech\acrobotics\code\jona\Acrobot/.pio/libdeps/main/LcdMenu/src/utils.h:31: multiple definition of `remove(char*, unsigned char, unsigned char)'; .pio\build\main\src\main.cpp.o:D:\9mediatech\acrobotics\code\jona\Acrobot/.pio/libdeps/main/LcdMenu/src/utils.h:31: first defined here
c:/users/daniel/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: .pio\build\main\lib4fd\libmenu.a(MenuClass.cpp.o): in function `mapProgress(unsigned short, long, long)':
D:\9mediatech\acrobotics\code\jona\Acrobot/.pio/libdeps/main/LcdMenu/src/utils.h:39: multiple definition of `mapProgress(unsigned short, long, long)'; .pio\build\main\src\main.cpp.o:D:\9mediatech\acrobotics\code\jona\Acrobot/.pio/libdeps/main/LcdMenu/src/utils.h:39: first defined here
c:/users/daniel/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: .pio\build\main\lib4fd\libmenu.a(MenuClass.cpp.o): in function `mapProgress(unsigned short, float, float)':
D:\9mediatech\acrobotics\code\jona\Acrobot/.pio/libdeps/main/LcdMenu/src/utils.h:44: multiple definition of `mapProgress(unsigned short, float, float)'; .pio\build\main\src\main.cpp.o:D:\9mediatech\acrobotics\code\jona\Acrobot/.pio/libdeps/main/LcdMenu/src/utils.h:44: first defined here
collect2.exe: error: ld returned 1 exit status
*** [.pio\build\main\firmware.elf] Error 1
```


To recreate:
main.cpp:
```cpp
#include "MenuClass.h"

MenuClass menuClass(0x27);

void setup(){
}

void loop(){
}
```

MenuClass.h:
```cpp

#ifndef MENUCLASS_H
#define MENUCLASS_H

#include <LcdMenu.h>

class MenuClass {
  public:
    MenuClass(uint8_t address);
  private:
    uint8_t address;
};

#endif
```

MenuClass.cpp:
```cpp
#include "MenuClass.h"

MenuClass::MenuClass(uint8_t address) {
  this->address = address;
}
```




### Version

3.3.2

### What environment are you seeing the problem in?

_No response_

### Board type

_No response_

### Relevant log output

_No response_

first note, let's not call classes `Class`. Like `MenuClass`, it should just be `Menu`. Because otherwise, in order to be consistent, you should rename `address` to `addressUint_8t`.

Regarding the error, it says "multiple definition of `substring". There's probably to `substring` functions in scope and the compiler can't tell which one to use, have you tried fully qualifying the path of the function?

--- a/examples/IntFloatValues/IntFloatValues.ino
+++ b/examples/IntFloatValues/IntFloatValues.ino
@@ -21,7 +21,7 @@
 #define BACKSPACE 8  // BACKSPACE
 #define CLEAR 46     // NUMPAD .
 
-// Declare the calbacks
+// Declare the callbacks
 void callback(uint16_t pos);
 
 char* intMapping(uint16_t progress) {

--- a/examples/List/List.ino
+++ b/examples/List/List.ino
@@ -2,7 +2,7 @@
  Menu Item List
 
  https://lcdmenu.forntoh.dev/examples/string-list
- 
+
 */
 
 #include <ItemList.h>
@@ -21,7 +21,7 @@
 #define BACKSPACE 8  // BACKSPACE
 #define CLEAR 46     // NUMPAD .
 
-// Declare the calbacks
+// Declare the callbacks
 void colorsCallback(uint16_t pos);
 void numsCallback(uint16_t pos);
 
@@ -73,7 +73,7 @@ void loop() {
         menu.back();
 }
 
-// Define the calbacks
+// Define the callbacks
 void colorsCallback(uint16_t pos) {
     // do something with the index
     Serial.println(colors[pos]);

--- a/src/utils.cpp
+++ b/src/utils.cpp
@@ -1,0 +1,50 @@
+#include "utils.h"
+
+#include "Constants.h"
+
+void substring(char* str, uint8_t start, uint8_t size, char* substr) {
+    strncpy(substr, str + start, size);
+    substr[size] = '\0';
+}
+
+void concat(const char* first, char second, const char* third, char* result) {
+    size_t len1 = strlen(first);
+    size_t len3 = strlen(third);
+
+    memcpy(result, first, len1);
+    result[len1] = second;
+    memcpy(result + len1 + 1, third, len3);
+    result[len1 + 1 + len3] = '\0';
+}
+
+void concat(const char* first, char second, char* result) {
+    size_t len1 = strlen(first);
+    memcpy(result, first, len1);
+    result[len1] = second;
+    result[len1 + 1] = '\0';
+}
+
+void concat(const char* first, const char* second, char* result) {
+    strcpy(result, first);
+    strcat(result, second);
+}
+
+void remove(char* str, uint8_t index, uint8_t count) {
+    uint8_t len = strlen(str);
+    if (index + count > len) {
+        count = len - index;
+    }
+    memmove(str + index, str + index + count, len - count - index + 1);
+}
+
+long mapProgress(uint16_t progress, long minValue, long maxValue) {
+    // Map the progress value to a new range (minValue to maxValue)
+    return map(progress, MIN_PROGRESS, MAX_PROGRESS, minValue, maxValue);
+}
+
+float mapProgress(uint16_t progress, float minValue, float maxValue) {
+    // Normalize the progress value and map it to the specified floating-point
+    // range
+    return static_cast<float>(progress) / MAX_PROGRESS * (maxValue - minValue) +
+           minValue;
+}

--- a/src/utils.h
+++ b/src/utils.h
@@ -1,49 +1,20 @@
-#pragma once
+#ifndef MenuUtils_H
+#define MenuUtils_H
+
 #include <Arduino.h>
 
-void substring(char* str, uint8_t start, uint8_t size, char* substr) {
-    strncpy(substr, str + start, size);
-    substr[size] = '\0';
-}
+void substring(char* str, uint8_t start, uint8_t size, char* substr);
 
-void concat(const char* first, char second, const char* third, char* result) {
-    size_t len1 = strlen(first);
-    size_t len3 = strlen(third);
+void concat(const char* first, char second, const char* third, char* result);
 
-    memcpy(result, first, len1);
-    result[len1] = second;
-    memcpy(result + len1 + 1, third, len3);
-    result[len1 + 1 + len3] = '\0';
-}
+void concat(const char* first, char second, char* result);
 
-void concat(const char* first, char second, char* result) {
-    size_t len1 = strlen(first);
-    memcpy(result, first, len1);
-    result[len1] = second;
-    result[len1 + 1] = '\0';
-}
+void concat(const char* first, const char* second, char* result);
 
-void concat(const char* first, const char* second, char* result) {
-    strcpy(result, first);
-    strcat(result, second);
-}
+void remove(char* str, uint8_t index, uint8_t count);
 
-void remove(char* str, uint8_t index, uint8_t count) {
-    uint8_t len = strlen(str);
-    if (index + count > len) {
-        count = len - index;
-    }
-    memmove(str + index, str + index + count, len - count - index + 1);
-}
+long mapProgress(uint16_t progress, long minValue, long maxValue);
 
-long mapProgress(uint16_t progress, long minValue, long maxValue) {
-    // Map the progress value to a new range (minValue to maxValue)
-    return map(progress, MIN_PROGRESS, MAX_PROGRESS, minValue, maxValue);
-}
+float mapProgress(uint16_t progress, float minValue, float maxValue);
 
-float mapProgress(uint16_t progress, float minValue, float maxValue) {
-    // Normalize the progress value and map it to the specified floating-point
-    // range
-    return static_cast<float>(progress) / MAX_PROGRESS * (maxValue - minValue) +
-           minValue;
-}
+#endif

